### PR TITLE
JDK18 bringup - add JAVA18 JPP pConfig, Access.exit & MajorVersion

### PR DESCRIPTION
**Pull request by JasonFengJ9**
Added `JPP pConfig JAVA18`;
Added `BCT_Java18MajorVersionShifted`;
Added `Access.exit(status)`;

With this PR, `-version` output for JDKnext `openj9-staging`:
```
openjdk version "18-internal" 2022-03-15
OpenJDK Runtime Environment (build 18-internal+0-adhoc.fengjcaibmcom.mergetmp)
Eclipse OpenJ9 VM (build master-d8104c46a0, JRE 18 Mac OS X amd64-64-Bit Compressed References 20210611_000000 (JIT enabled, AOT enabled)
OpenJ9   - d8104c46a0
OMR      - 99f008c99
JCL      - da7073a7c4b based on jdk-9+99)
```
The JCL version is a known issue due to merging script.

close https://github.com/eclipse-openj9/openj9/issues/12930

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

File: jcl/src/java.base/share/classes/java/lang/Access.java
Comment (keithc-ca) [inline jcl/src/java.base/share/classes/java/lang/Access.java:515]: I don't think this belongs within the `OPENJDK_METHODHANDLES` section.

File: runtime/jcl/common/sun_misc_Unsafe.cpp
Comment (keithc-ca) [inline runtime/jcl/common/sun_misc_Unsafe.cpp:1104]: I think you should leave things like this alone. If we decide to clean up, it should be done separately.

File: runtime/oti/j9nonbuilder.h
Comment (keithc-ca) [inline runtime/oti/j9nonbuilder.h:2057]: Removing old versions should be done in a separate PR (and would then also remove the macros for 2, 5, 6, 7 and 9 as well).

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: runtime/jcl/common/sun_misc_Unsafe.cpp
Comment (JasonFengJ9) [reply runtime/jcl/common/sun_misc_Unsafe.cpp:1104]: I feel these obsoleted defines don't have much references, could be replaced in a straightforward way for cleaner code.
If you prefer a separated PR for such changes, will open one then.

File: jcl/src/java.base/share/classes/java/lang/Access.java
Comment (keithc-ca) [reply jcl/src/java.base/share/classes/java/lang/Access.java:515]: Ok, I had forgotten about that detail.

File: runtime/oti/j9nonbuilder.h
Comment (JasonFengJ9) [reply runtime/oti/j9nonbuilder.h:2057]: > remove the macros for 2, 5, 6, 7 and 9 as well

Those versions have more references, and the cleanup is not straightforward. I think those area need to be examined more carefully which I would prefer in another PR after the easy cleanup PR.

Comment (keithc-ca) [reply runtime/oti/j9nonbuilder.h:2057]: I think that cleanup should remove `J2SE_VERSION` altogether in favour of using `JAVA_SPEC_VERSION`. There's no longer a reasonable use-case where those tests would disagree.

Comment (JasonFengJ9) [reply runtime/oti/j9nonbuilder.h:2057]: That will be an even larger refactoring, I attempted couple of years ago at https://github.com/eclipse-openj9/openj9/pull/5370.
Would suggest do it in another separated PR.

File: runtime/jcl/common/sun_misc_Unsafe.cpp
Comment (JasonFengJ9) [reply runtime/jcl/common/sun_misc_Unsafe.cpp]: Created https://github.com/eclipse-openj9/openj9/pull/12935

**Comment by pshipton** [comment]
> The JCL version is a known issue due to merging script.

If the merging script is wrong, can't we back out the bad change for now?

**Comment by pshipton** [comment]
Also, we can update closed/openjdk-tag.gmk to have the correct version.

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
> Also, we can update closed/openjdk-tag.gmk to have the correct version.
That would happen in the extensions repo.

> If the merging script is wrong, can't we back out the bad change for now?
I suggest we merge my proposed fix https://github.com/adoptium/temurin-build/pull/2653 (again, a different repo).

**Comment by pshipton** [comment]
Yes, I realize it's a change in a different repo, but I'm saying we should fix it today, not wait around for the temurin-build fix to be merged. None of us are committers on that repo, so we have no control over when/if that will be merged. I'm happy to take the same change to our repo though.

**Comment by JasonFengJ9** [comment]
> If the merging script is wrong, can't we back out the bad change for now?

The PR in question has been reverted, next merging run should fix itself, otherwise, will submit a manual fix.

**Comment by pshipton** [comment]
In terms of this PR, it seems good to me. We can run PR (compile) testing against the openj9-staging branch, if Keith agrees.

**Comment by keithc-ca** [comment]
Jenkins compile zlinux jdknext depends ibmruntimes/openj9-openjdk-jdk#openj9-staging

**Comment by JasonFengJ9** [comment]
The tag has been updated properly:
```
OPENJDK_TAG := jdk-17+18
```
https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/c931952254b14c73c389a53413e4b702e303df73/closed/openjdk-tag.gmk#L1

**Comment by keithc-ca** [comment]
@pshipton I'll let you merge this, assuming you approve.

**Review by pshipton — APPROVED** [review]

**Pull request merged**
